app/src/main/java/com/example/android/codelabs/paging/data/GithubRepository.kt

```diff
@@ -17,60 +17,75 @@
 package com.example.android.codelabs.paging.data
 
 import android.util.Log
-import androidx.lifecycle.MutableLiveData
 import com.example.android.codelabs.paging.api.GithubService
 import com.example.android.codelabs.paging.api.searchRepos
-import com.example.android.codelabs.paging.db.GithubLocalCache
+import com.example.android.codelabs.paging.model.Repo
 import com.example.android.codelabs.paging.model.RepoSearchResult
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.channels.ConflatedBroadcastChannel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.asFlow
 
 /**
  * Repository class that works with local and remote data sources.
  */
-class GithubRepository(
-    private val service: GithubService,
-    private val cache: GithubLocalCache
-) {
+@FlowPreview
+@ExperimentalCoroutinesApi
+class GithubRepository(private val service: GithubService) {
+
+    // keep the list of responses
+    private val inMemoryCache = ConflatedBroadcastChannel<RepoSearchResult>()
```

Comment:
This seems a little strange to me, having a channel like this as a cache, is this an idiomatic way to use it? (I'm just not familiar enough to say)

Reply:
@manuelvicnt said this is ok.